paddle/fluid/inference/api/mkldnn_analysis_predictor.cc

```diff
@@ -0,0 +1,178 @@
+  } else if (!config_.prog_file.empty() && !config_.param_file.empty()) {
+    // All parameters are saved in a single file.
+    // The file names should be consistent with that used
+    // in Python API `fluid.io.save_inference_model`.
+    inference_program_ = paddle::inference::Load(
+        executor_.get(), scope_.get(), config_.prog_file, config_.param_file);
+  } else {
+    LOG(ERROR) << "fail to load inference model.";
+    return false;
+  }
+
+  if (IsMKLDNNSetOn()) {
+    LOG(INFO) << "MKL-DNN enabled";
+    config_.use_mkldnn = true;
+    executor_->EnableMKLDNN(*inference_program_);
+  }
+
+  OptimizeInferenceProgram();
+  ctx_ = executor_->Prepare(*inference_program_, 0);
+
+  VLOG(5) << "to create variables";
+  PADDLE_ENFORCE(scope_.get());
+  executor_->CreateVariables(*inference_program_,
+                             sub_scope_ ? sub_scope_ : scope_.get(), 0);
+  // Get the feed_target_names and fetch_target_names
+  PrepareFeedFetch();
+  return true;
+}
+
+void MKLDNNAnalysisPredictor::OptimizeInferenceProgram() {
+  LOG(INFO) << "== optimize begin ==";
+  FLAGS_IA_enable_ir = config_.enable_ir_optim;
+  FLAGS_IA_enable_tensorrt_subgraph_engine = false;
+  FLAGS_IA_output_storage_path = "";  // Don't output the model.
+  // Analyze inference_program
+  if (!config_.model_dir.empty()) {
+    analysis_argument().fluid_model_dir.reset(new std::string(config_.model_dir));
+  } else {
+    PADDLE_ENFORCE(
+        !config_.param_file.empty(),
+        "Either model_dir or (param_file, prog_file) should be set.");
+    PADDLE_ENFORCE(!config_.prog_file.empty());
+    analysis_argument().fluid_model_program_path.reset(
+        new std::string(config_.prog_file));
+    analysis_argument().fluid_model_param_path.reset(new std::string(config_.param_file));
+  }
+  analysis_argument().origin_program_desc.reset(
+      new ProgramDesc(*inference_program_->Proto()));
+
+  PADDLE_ENFORCE(config_.ir_mode == AnalysisConfig::IrPassMode::kInclude,
+		 "Only kInclude is supported with MKLDNNAnalysisPredictor.");
+  if (config_.use_mkldnn)
+    MKLDNNAnalyzer().SetIrPasses(config_.ir_mkldnn_passes).Run(&analysis_argument());
+  else
+    MKLDNNAnalyzer().SetIrPasses(config_.ir_passes).Run(&analysis_argument());
+
+  CHECK(analysis_argument().transformed_program_desc);
+  VLOG(5) << "to prepare executor";
+  inference_program_.reset(
+      new framework::ProgramDesc(*analysis_argument().transformed_program_desc));
+  if (analysis_argument().Has(framework::ir::kParamScopeAttr)) {
+    // Update scope.
+    scope_.reset(
+        analysis_argument().Release<framework::Scope>(framework::ir::kParamScopeAttr));
+  }
+  LOG(INFO) << "== optimize end ==";
+}
+
+template <>
+std::unique_ptr<PaddlePredictor> CreatePaddlePredictor<
+    MKLDNNAnalysisConfig, PaddleEngineKind::kAnalysis>(
+		    const MKLDNNAnalysisConfig& config) {
+  VLOG(3) << "create MKLDNNAnalysisConfig";
+  if (config.use_gpu) {
```

Comment:
line 153-169 could be removed since `config.use_gpu` is 0 when use MKLDNNconfig

Reply:
Done.